sklearn_numba_dpex/kmeans/engine.py

```diff
@@ -310,9 +332,33 @@ def _validate_data(self, X, reset=True):
                 ):
                     raise NotSupportedByEngineError from type_error
 
-    def _check_sample_weight(self, sample_weight, X):
+    def _check_is_accepted_X(self, X):
+        if X is not self._X_accepted:
+            raise RuntimeError(
+                "The object that was passed to the engine to query its compatibility "
+                "is different from the object that was given in downstream methods."
+            )
+
+    def _check_is_accepted_sample_weight(self, sample_weight):
+        if sample_weight is not self._sample_weight_accepted:
+            raise RuntimeError(
+                "The object that was passed to the engine to query its compatibility "
+                "is different from the object that was given in downstream methods."
+            )
+
+        # When sample_weight is None, the call to `_check_sample_weight` is delayed
+        # until now because, because the array of `ones` that is created is only
+        # necessary for engine methods that actually make use of `sample_weight` and
+        # call `_check_is_accepted_sample_weight`.
+        # Methods that don't use `sample_weight` still pass `sample_weight=None` to
+        # `accepts` but doesn't need to create the corresponding array.
```

Comment:
```suggestion
        # When sample_weight is None, the call to `_check_sample_weight` is delayed
        # until now because the array of `ones` that is created is only necessary
        # for engine methods that actually make use of `sample_weight` and call 
        # `_check_is_accepted_sample_weight`.
        # Methods that don't use `sample_weight` still pass `sample_weight=None` to
        # `accepts` but doesn't need to create the corresponding array.
```